### ISSUE TITLE
Components and Containers

### ISSUE THREAD

**Issue by twishasaraiya** [UdacityMobileWebScholarship/guess-quote#56]
Static data will be passed in for now and redux configuration will be done later.

**Basic Flow**
1. Landing Page from where the user gets 2 options: _Start Playing_ or _Login_
2. Depending on the user choice he goes to step 3 or 4
3. Login/Signup then move to step 4
4. Quote and Options then depending on answer respective component gets added to the page
5. If user logouts of game, all data will be lost

**Components(props)**
1. Quote and Options(quote,options[])
2. Score(score)
3. Timer(time left)
4. HowToPlay
5. CorrectAnswer
6. Wrong Answer

**Containers**
1. Quote and Options
2. Score
3. Timer

Besides these, there is **Login/SignUp** Component. You can refer this [Issue](https://github.com/UdacityMobileWebScholarship/guess-quote/issues/35)

Please let me know your suggestions and opinions here.

**Comment by PN1019** [comment]
what about progress component?? I think it is required to display progression like

> question 1 of 20

 ??
For this, 
it should take two properties 

- `currentQuestionIndex`

- ` numberOfQuestions`

**Comment by twishasaraiya** [comment]
@PN1019 Yes it could be done but in our current UI designs we have not added progress bar because the number questions (like you said 20) is still debatable and not decided.
Let me know your opinions.

**Comment by PN1019** [comment]
@twishasaraiya it doesn't matter how many numbers of questions you are planning to move with it totally depends on your call well since you are specifying that UI designs don't have progress bar as of now  this can also be included in next version  if don't want to make any changes in UI now..but I think it will good to go with Progression even now since this will engage users in quiz...

Also, how are you planning to move to next question?via NextButton or Timer ?? I think NextButton component should be there since many players can answer the question before the time they need not have to wait for the timer to finish off..!!

I don't understand why there is the need for two answer components since there will be only one correct answer? I think we can go with the radio button for that !!!

**Comment by twishasaraiya** [comment]
@PN1019 I think the progress bar can be included in the next version since now our focus to make it a PWA first.

Also, I agree with you that there should be a  `Next Button`. Yes, we will be having the next Button once the user has clicked an option. I have updated the component section have a look.

I am assuming you are talking about `CorrectAnswer` and `WrongAnswer`. What this means this that there will a text(like `Better luck next time`) and an image(like an animated cross or tick mark) component added to the page depending on the answer. Let me know what you think about it.

**Comment by PN1019** [comment]
@twishasaraiya  yes I am talking about **CorrectAnswer** and **WrongAnswer** only.
Okay, got it you want to handle both things separately go with text and image thing you mentioned above ...but this can be done via a function like `checkanswer` in the same component isn't it ?? Any other reason for creating two separate components?? IDK if i am not wrong both component will do same thing but with different response ?? so why we need two separate components to perform this action??Although ,as per you we are also creating **_Options_**  component which can handle all options  and we can move further with **_Answer_**  component...What say??

**Comment by twishasaraiya** [comment]
Okay let's have one `component` called *AnswerComponent* which display correct response based on `checkAnswer` method.

**Comment by drenther** [comment]
The design assets only seem to have mobile UI mockups. Any thoughts or opinions on how it should scale on bigger screen sizes?

**Comment by twishasaraiya** [comment]
@drenther Thanks for asking since I guess many people would have the same doubt.I think @ayushdev and @VarunDevPro will be able to explain better since they have designed the mockups.

**Comment by ayushdev** [comment]
@drenther We can use custom media queries for small, mid, and large screen sizes. As the containers inside pages are very simple, I think they will be handled easily.

**Comment by PN1019** [comment]
@twishasaraiya  I would like to work on this let me know what I can take up ?? Anyone already working on components??

**Comment by drenther** [comment]
@ayushdev I did not mean the implementation details. Of course, we will use custom media queries. I meant how It should look on tablet and desktops should also be represented in designs for the sake of seamless implementation when different people implement different components. So that the design choices won't be made by everyone on their own but inferred from the design mockups. 

To look at it more practically - 
Suppose we talk about the drawer menu. That is something useful for the mobile screen but not the desktop screen. Desktops screens have enough space to accommodate those options without a hamburger menu option as well and decisions like how large should the options and quotes sections be on tablet or desktop.

Any thoughts?

**Comment by twishasaraiya** [comment]
@PN1019 So I am assigning this issue to you. You can work on the main `QuoteAndOptions` Component and the buttons Component as well. Nobody has taken up anything yet but 2 people were interested. I told them to add a comment here first. So far they haven't.
We need to complete this by end of the week, so if you need any help please let me know.

**Comment by suhasbansude** [comment]
I am interesting to work on QuoteAndOptions component.Can i start work on this.

**Comment by twishasaraiya** [comment]
@suhasbansude Since I have asked @PN1019 about it. Let's wait for the reply. If  he is not working on QuoteandOption then you can definitely take it up.
You could also work on other components and make a PR. Let me know what you are taking up.

**Comment by PN1019** [comment]
@suhasbansude if you wanna take QuoteAndOptions take that up...it just fine with.... @twishasaraiya.... thanks...I will work on by button components you mean NextButton ??? If that's the case I am good to go with....please let me know

**Comment by twishasaraiya** [comment]
Okay so @suhasbansude you are taking up
`QuotenOptions` component and container.
@PN1019 you will be working on `nextButton` and `Answer` so that suhas can call these component when corresponding  event occurs. 

If you have any doubts please free to ask them.

**Comment by skarthikeyan96** [comment]
@twishasaraiya  Can I work on the HowToPlay Instruction Component

**Comment by twishasaraiya** [comment]
@skarthikikeyan96 Yes you can work on it.

**Comment by PN1019** [comment]
@twishasaraiya  you haven't specified any testing guidelines for react components??

**Comment by twishasaraiya** [comment]
I am extremely sorry for late reply as I was busy with interviews.
  I just want to add here that we want the contributors to submit a working PR and a screenshot along with it. We currently don't have any plans of adding Travis since it is required to be setup by repository owner. So currently we will be checking your PR ourselves before merging. 

I request you all to submit PR by **Sunday**   or before as soon as possible.

**Comment by twishasaraiya** [comment]
Yes please @anantoghosh you can work on it.

**Comment by AromaR** [comment]
Hi, will be working on the Score Component.

**Comment by Rohanhacker** [comment]
I'll be working on timer and score component

**Comment by twishasaraiya** [comment]
@Rohanhacker I m sorry but  others are already working on it.

**Comment by Rohanhacker** [comment]
damm

**Comment by Rohanhacker** [comment]
is anything available ?

**Comment by twishasaraiya** [comment]
Nope as of now. You could work on homepage if you want with @ayushdev.
